internal/multistorage/object.go

```diff
@@ -5,16 +5,38 @@ import (
 	"github.com/wal-g/wal-g/pkg/storages/storage"
 )
 
-var _ storage.Object = multiObject{}
+// MultiObject is any storage.Object that can tell the name of the storage in which it is stored.
+type MultiObject interface {
+	storage.Object
+	GetStorage() string
+}
+
+var _ MultiObject = multiObject{}
 
+// multiObject is an internal implementation of MultiObject that is provided from multistorage.Folder methods
+// instead of the simple storage.Object.
 type multiObject struct {
 	storage.Object
-	storageName string
+	StorageName string
 }
 
+func (mo multiObject) GetStorage() string {
+	return mo.StorageName
+}
+
+// GetStorage provides the name of the storage where the object is stored. If the object can't tell the storage name on
+// its own, provides "default".
 func GetStorage(obj storage.Object) string {
-	if mo, ok := obj.(multiObject); ok {
-		return mo.storageName
+	// TODO: Don't rely on different storage.Object implementations here (issue #TODO). Possible solutions:
```

Comment:
@usernamedt  Is github issues a suitable place to put such refactoring task?

Reply:
What if we just switch to the `multistorage.Object` in our delete handler? To avoid all this logic with guessing object types